src/devices/cpu/drcbex64.cpp

```diff
@@ -1482,9 +1591,16 @@ void drcbe_x64::op_nop(Assembler &a, const instruction &inst)
 //  op_break - process a BREAK opcode
 //-------------------------------------------------
 
+void break_into_debugger()
+{
+	static const char *message = "break from drc";
+	osd_break_into_debugger(message);
+}
+
 void drcbe_x64::op_break(Assembler &a, const instruction &inst)
 {
-	smart_call_r64(a, (x86code *)(uintptr_t)&osd_break_into_debugger, rax);
+	a.int3();
+	smart_call_r64(a, (x86code *)(uintptr_t)&break_into_debugger, rax);
 }
```

Comment:
You shouldn’t need the trampoline function, e.g. see what `op_callc` does:
```c++
void drcbe_x64::op_callc(Assembler &a, const instruction &inst)
{
    // validate instruction
    assert(inst.size() == 4);
    assert_any_condition(inst);
    assert_no_flags(inst);

    // normalize parameters
    const parameter &funcp = inst.param(0);
    assert(funcp.is_c_function());
    be_parameter paramp(*this, inst.param(1), PTYPE_M);

    // skip if conditional
    Label skip = a.newLabel();
    if (inst.condition() != uml::COND_ALWAYS)
        a.short_().j(X86_NOT_CONDITION(inst.condition()), skip);                        // jcc   skip

    // perform the call
    mov_r64_imm(a, Gpq(REG_PARAM1), (uintptr_t)paramp.memory());                        // mov   param1,paramp
    smart_call_r64(a, (x86code *)(uintptr_t)funcp.cfunc(), rax);                        // call  funcp

    // resolve the conditional link
    if (inst.condition() != uml::COND_ALWAYS)
        a.bind(skip);                                                               // skip:
}
```

Also, missing one `const` for the pointer.

You should be able to do something like:
```c++
    static char const *const message = "break from drc";
    mov_r64_imm(a, Gpq(REG_PARAM1), (uintptr_t)message);
    smart_call_r64(a, (x86code *)(uintptr_t)&break_into_debugger, rax);
```

Reply:
I mentioned it in a different comment, but that's what I was doing for x64 but x86 was doing something a bit different so I thought it'd be easier to implement them the same way and leave it up to the compiler.

Reply:
Sorry, should be:
```c++
    static char const *const message = "break from drc";
    mov_r64_imm(a, Gpq(REG_PARAM1), (uintptr_t)message);
    smart_call_r64(a, (x86code *)(uintptr_t)&osd_break_into_debugger, rax);
```

Reply:
Ok. https://github.com/mamedev/mame/pull/13108/commits/7da0d8e2bf0ec88ce2422f0bee3f4f52d21c639f